extensions/qute/deployment/src/main/java/io/quarkus/qute/deployment/QuteProcessor.java

```diff
@@ -1442,16 +1437,48 @@ static boolean processHints(TemplateAnalysis templateAnalysis, List<String> help
                     }
                 }
             } else if (helperHint.startsWith(SetSectionHelper.Factory.HINT_PREFIX)) {
-                Expression valueExpr = findExpression(helperHint, SetSectionHelper.Factory.HINT_PREFIX, templateAnalysis);
-                if (valueExpr != null) {
-                    Match valueExprMatch = generatedIdsToMatches.get(valueExpr.getGeneratedId());
-                    if (valueExprMatch != null) {
-                        match.setValues(valueExprMatch.clazz, valueExprMatch.type);
+                setMatchValues(match, findExpression(helperHint, SetSectionHelper.Factory.HINT_PREFIX, templateAnalysis),
+                        generatedIdsToMatches, index);
+            }
+        }
+        return false;
+    }
+
+    private static void setMatchValues(Match match, Expression valueExpr, Map<Integer, Match> generatedIdsToMatches,
+            IndexView index) {
+        if (valueExpr != null) {
+            if (valueExpr.isLiteral()) {
+                Object literalValue;
+                try {
+                    literalValue = valueExpr.getLiteralValue().get();
+                } catch (InterruptedException | ExecutionException e) {
+                    literalValue = null;
+                }
+                if (literalValue == null) {
+                    match.clearValues();
+                } else {
+                    if (literalValue instanceof Boolean) {
+                        match.setValues(index.getClassByName(DotNames.BOOLEAN), Types.box(Primitive.BOOLEAN));
+                    } else if (literalValue instanceof String) {
+                        match.setValues(index.getClassByName(DotNames.STRING),
+                                Type.create(DotNames.STRING, org.jboss.jandex.Type.Kind.CLASS));
+                    } else if (literalValue instanceof Integer) {
+                        match.setValues(index.getClassByName(DotNames.INTEGER), Types.box(Primitive.INT));
+                    } else if (literalValue instanceof Long) {
+                        match.setValues(index.getClassByName(DotNames.LONG), Types.box(Primitive.LONG));
+                    } else if (literalValue instanceof Double) {
+                        match.setValues(index.getClassByName(DotNames.DOUBLE), Types.box(Primitive.DOUBLE));
+                    } else if (literalValue instanceof Float) {
```

Comment:
Should `Character` be tested too?

Reply:
Nope, only `String` literals are supported in templates, i.e. `b` is converted to `String` of value `b`.